### ISSUE TITLE
Low PCC for negative values in exp

### ISSUE THREAD

**Issue by boris-drazic** [tenstorrent/tt-metal#3612]
`exp` OP has issues when input values are very negative, e.g., numbers from -1000 to -inf.
This is most commonly seen with attention masks in LLM, where a mask with -inf ends up in `softmax`, which uses `exp` in computation, resulting in bad PCC for `softmax`.

As a workaround, using -100 instead of -inf in attention mask usually results in good PCC for `softmax`.

**Comment by jvasilje** [comment]
@eyonland please add to eltwise boad

**Comment by eyonland** [comment]
@yan-zaretskiy, I think this might be a limitation of the underlying llk and may need to be included on #9702.

**Comment by yan-zaretskiy** [comment]
This indeed could be an LLK issue. Here I compared the $\exp(-x)$ results from `numpy` using `float32`, CUDA with `bfloat16` and `tt-metal` with `bfloat16`. `tt-metal` drops to zero before CUDA, which could explain the PCC effect. You can also see that we are slightly off for $x \in [60, 80]$. The real question is the $L_2$ error though. I suspect the PCC unfairly penalizes these minuscule errors just because they behave as numerical noise and can thus reduce the linear correlation.

![exp](https://github.com/user-attachments/assets/b822405b-fd9f-4251-89d0-098912cd5473)

**Comment by yan-zaretskiy** [comment]
I suggest we look at the actual $L_2$ norm of the error between `tt-metal` and `pytorch` to see if there are genuine errors in the softmax somewhere. I don't trust the PCC measure here.

**Comment by prajaramanTT** [comment]
@boris-drazic Is this still an open issue ? If not, can you please close this ? Thanks

**Comment by nvelickovicTT** [comment]
This is covered more holistically in #19821. Closing this instance.
